### DISCUSSION TITLE
Auto formatting code using clang-format

### DISCUSSION THREAD

**Discussion by cflaviu** [managarm/managarm#306] [Ideas and RFCs]
Applying a formatting rules file over all source contributions, it will reduce the time editing files and all the code will be standardized.
The rules could be set based on https://releases.llvm.org/11.0.0/tools/clang/docs/ClangFormatStyleOptions.html resulting a .clang-format file stored in .git folder of project. The all staged files could be formatted calling git clang-format command.

**Comment by ArsenArsen** [comment]
We've spoken about this before, the concern brought up is that there's a possibility that it's too aggressive, leading to technically correct but practically unreadable formatting in some spaces, with no (clean) workaround.

**Reply by cflaviu** [reply]
I had same concern in the past but after using the tool, I love it. I can code fast without care about formatting and at save time the file will perfectly formatted.

**Reply by cflaviu** [reply]
We could try at least.

**Reply by 64** [reply]
I think the benefits outweigh the downsides here a lot (code formatting conversations seem to pop up quite frequently), someone just needs to knuckle down and write one

another problem is that the style  isn't consistent across all parts of the code, e.g `mlibc` and `frigg` seems to use snake case for everything

**Reply by cflaviu** [reply]
@64 Clang-format doesn't change the case of identifiers. Clang-tidy is the tool that would complain about identifier case.

**Reply by 64** [reply]
Fair enough, though it's more than just identifier case that is different

**Comment by cflaviu** [comment]
I created a .clang-format file matching as much as possible the current code although the coding style is not exactly the same in every source file. Please try and adapt if necessary this file https://gist.github.com/cflaviu/e95dd1272287b561c77b0b573092f425 .

**Reply by Dennisbonke** [reply]
As mentioned on the Discord, you should double check that it conforms to the [coding standard](https://docs.managarm.org/handbook/contributing/coding-style.html) we created.

**Reply by Dennisbonke** [reply]
I took a quick look at it and I see several macro’s that seem related to QT and Boost, I think those can be removed. Furthermore, I’m unsure if the file actually follows the coding style. Can you double check that before I try it myself?
